### ISSUE TITLE
Module fails to load after a copy+paste operation

### ISSUE THREAD

**Issue by SMB7** [rerrahkr/BambooTracker#137]
This issue (judging by the text in question) seems similar to #126; however, it doesn't happen after opening an included demo module, and doesn't continue creating errors.

Basically, I made an arrangement of a song without using the ext. CH3 option. I decided I wanted to try making it with ext. CH3 to free up an SSG channel for more drums, but BambooTracker wouldn't let me enable the option on an existing song, so I copied it over to a new song. Then I began moving the contents of the module to the right so that they were in the correct columns. 
However, one of these paste operations caused BambooTracker to crash, and attempting to re-open the module gives this error:
![image](https://user-images.githubusercontent.com/22063427/59854851-ecad3b80-9330-11e9-91b1-225491c4f31d.png)
(interestingly, the song still seems to have loaded as you can see by the channels behind the error window)
Closing this error then causes BambooTracker to default back to a new module.

The module in question: https://drive.google.com/open?id=1chuk3mcjcdnXSfFzOYlbF3eVYwxvzS8G
I have work now, but I will check when I can if the .bak of the module is still usable or has also been corrupted.

**Comment by jpcima** [comment]
This is your module working again after some little cleanup and resaving. [EoSD-2.btm.gz](https://github.com/rerrahkr/BambooTracker/files/3311012/EoSD-2.btm.gz)
It was created on a newer version 0.2.1, to which an update is recommended.

Apparently a pattern step index out of bounds was existing in the file.

**Comment by SMB7** [comment]
Thank you for this, however, the program (now updated to 0.2.1) still crashes when attempting to play song 1 in the module (though editing appears to work properly, and the module no longer becomes corrupt afterward; additionally song 0 plays without issue). On the other hand, the backup of the original module is still working, so thankfully this issue does not appear *too* serious, though some work was lost in copying song 0 to song 1 and I'd rather avoid having this happen again.

**Comment by SMB7** [comment]
After fiddling around a bit more I believe that the crash is occurring due to a volume slide on the ext. CH3 operators, as it only stopped happening when that effect was removed. Placing the same volume slide effect on all four operators (initially it was only on operators 3 and 4) did not prevent the crash, and the crash also occurred when the volume slide was only present on operators 1 and 2.

I'm unsure of the relevance of this crash to the original module corruption issue, which I have been unsuccessfully trying to replicate. It is possible that this crash was responsible for the original corruption in the previous version and that it has been fixed in 0.2.1, but I do not recall for sure the position the tracker was in at that time.

**Comment by rerrahkr** [comment]
Thanks for several reports.
As @jpcima mentioned, it caused by a wrong step index in song 01 track 6 (FM3-OP4) pattern 04. I checked copied pattern (maybe song 00 FM6 pattern 04) I found something is wrong about the effect entries in head steps of this pattern. This may have some effect on the paste.
![pattern](https://user-images.githubusercontent.com/26477882/59916841-3a25c780-945b-11e9-8eea-72daafeb66c5.png)

For volume slide effect, I fixed it at d39e2b7710c3128f2ede582e7b7ffec80f097f2f after I found accessing out of the range of the array.
